### ISSUE TITLE
Feature Request: Enable .swiftlint.yml to include a remote rules configuration

### ISSUE THREAD

**Issue by 72A12F4E** [realm/SwiftLint#2729]
It would be really useful in our organization to allow a `.swiftlint.yml` to be remotely included from another `.swiftlint.yml` config. We maintain several iOS / Swift projects, and keeping coding standards synchronized across all of them is difficult. This feature would enable us to create a company-wide config and have each project import that configuration, and we could easily update the config and apply new rules to the portfolio of projects we maintain.

Here is an example of how this could work.

**.swiftlint.yml**
```yaml
parent_config: https://foo.com/path/to/another/.swiftlint.yml

#...
```

What do y'all think?

**Comment by guzhenhuaGitHub** [comment]
So we can use one .swiftlint.yml in several projects. expecting...

**Comment by jpsim** [comment]
I like this, but a few details would need to be figured out:

* What do you do when there's poor to no network connectivity?
* Should SwiftLint make network calls for every command invocation? If not, how should these remote configs be cached?
* Would these external configs be versioned, if so how?
* Would it be possible to import more than one parent configuration?
* Would this feature apply equally to local configurations as well as remote ones?

**Comment by 72A12F4E** [comment]
What do you do when there's poor to no network connectivity?

* I think the job should surface an error / warning. This could be mitigated by caching the config on repeated invocations.

Should SwiftLint make network calls for every command invocation? If not, how should these remote configs be cached?

* I think it should attempt to get a fresh config every time, maybe there are config values to error/warn if a config is unavailable. The configs could be cached in the project root or somewhere like that.

Would these external configs be versioned, if so how?

* Thats a good question. I assume we could hash a config and use that to figure out if they have updated.

Would it be possible to import more than one parent configuration?

* I think that would make this more complicated than it needs to be.

Would this feature apply equally to local configurations as well as remote ones?

* There would have to be some sort of conflict resolution strategy for favoring the remote config or the local config... Favoring one or the other could be useful in different circumstances.

**Comment by jpsim** [comment]
Your suggestions sound reasonable. Do you need any help to get started on this @72A12F4E?

**Comment by bielikb** [comment]
@72A12F4E I would like to follow up on these efforts, would you like to collab in providing a solution? 
Have you already kicked-off some work in these regards?

**Comment by fredpi** [comment]
There's still an active PR addressing this: https://github.com/realm/SwiftLint/pull/3058

**Comment by fredpi** [comment]
This is now implemented in #3058 and will be available in the next release!
